src/context/blocked.rs

```diff
@@ -1,18 +1,151 @@
-use std::ptr;
+use crate::redismodule::{AUTH_HANDLED, AUTH_NOT_HANDLED};
+use crate::{raw, Context, ValkeyError, ValkeyString};
+use std::os::raw::{c_int, c_void};
 
-use crate::raw;
-use crate::Context;
+type AuthReplyCallback<T> =
+    fn(&Context, ValkeyString, ValkeyString, T) -> Result<c_int, ValkeyError>;
+type FreePrivDataCallback<T> = fn(&Context, T);
 
-pub struct BlockedClient {
+pub struct BlockedClient<T = ()> {
     pub(crate) inner: *mut raw::RedisModuleBlockedClient,
+    auth_reply_callback: Option<AuthReplyCallback<T>>,
```

Comment:
By adding this `auth_reply_callback: Option<AuthReplyCallback<T>>,` to this structure, we are coupling the `BlockedClient` structure to Authentication which does not seem right since there are other modes of blocking: 
1. General: BlockClient
2. Keys: BlockClientOnKeys


We could use an enum here which has variant for each type of ReplyCallback.

This will allow extending this interface to support blocking other modes